cmd/snap/cmd_run.go

```diff
@@ -91,6 +92,42 @@ and environment.
 		}, nil)
 }
 
+func maybeWaitForSecurityProfileRegeneration() error {
+	// check if the security profiles key has changed, if so, we need
+	// to wait for snapd to re-generate all profiles
+	mismatch, err := interfaces.SystemKeyMismatch()
+	if err != nil {
+		// something went wrong with the system-key compare, there
+		// is nothing we can do but continue and hope for the best
+		logger.Debugf("SystemKeyMismatch returned an error: %v", err)
+		return nil
+	}
+	if !mismatch {
+		return nil
+	}
+
+	// We have a mismatch, try to connect to snapd, once we can
+	// connect we just continue because that means snapd has
+	// generated new profiles.
+	//
+	// connect timeout is 5s
```

Comment:
are we sure 5s is enough on something slow?

Reply:
I clarified the comment, my reading of the client.doSync is that it waits for up to 5s to connect to snapd. I hope the updated comment makes this clear.